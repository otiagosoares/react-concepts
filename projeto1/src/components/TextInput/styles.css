.text-input{
    width: 100%;
    font-size:  2em;
    padding: 5px 10px;
}
