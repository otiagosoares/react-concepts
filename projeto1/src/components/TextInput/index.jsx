/* eslint-disable react/prop-types */
import React from 'react';
import P from 'prop-types';
import './styles.css';

const TextInput = ({ searchValue, handleChange }) => {
  return (
    <>
      <input
        type="search"
        className="text-input"
        value={searchValue}
        onChange={handleChange}
        placeholder="Type your search"
      />
    </>
  );
};

export default TextInput;

TextInput.propTypes = {
  searchValue: P.string.isRequired,
  handleChange: P.func.isRequired,
};
