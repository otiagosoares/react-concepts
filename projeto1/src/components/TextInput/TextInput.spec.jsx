import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TextInput from '.';
import userEvent from '@testing-library/user-event';

describe('<Posts />', () => {
  it('should call handleChange function on each kew pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue="valor qualquer" />);
    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';
    userEvent.type(input, value);
    expect(input.value).toBe('valor qualquer');
    //testa o numero de vezes que a tecla eh precionadada chamando a fn
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'testing...'} />);
    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input).toBeInTheDocument();
    expect(input.value).toBe('testing...');
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue={'testing...'} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
