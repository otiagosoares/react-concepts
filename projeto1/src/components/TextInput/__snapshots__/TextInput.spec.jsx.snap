// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type your search"
  type="search"
  value="testing..."
/>
`;
